### PR TITLE
C++ guidelines: update `_detail` namespace naming (missed occurrence)

### PR DESCRIPTION
**Pull request by antkmsft**

### CODE REVIEW

**Review by RickWinter — APPROVED** [review]
Nice catch

File: docs/cpp/introduction.md [hunk 1]
```diff
@@ -175,7 +175,7 @@ The C++ SDK is designed for synchronous api calls.
 
 {% highlight cpp %}
 namespace Azure { namespace Group { namespace Service {
-namespace _details {
+namespace _detail {
```
Comment (ahsonkhan) [inline docs/cpp/introduction.md:178]: Can you update other instances of "Details" on this page?

There are a couple here:
https://github.com/Azure/azure-sdk/blob/2144c42406447cc6637f877e7b6899af52c5d38a/docs/cpp/introduction.md#namespaces-cpp-namespace-naming

**Review by ahsonkhan — APPROVED** [review]
Consider fixing other mentions in this markdown file.

File: docs/cpp/introduction.md (hunk 1 above)
Comment (ahsonkhan) [reply docs/cpp/introduction.md:178]: Specifically, this section:
https://guidelinescollab.github.io/azure-sdk/cpp_design.html#cpp-design-naming-namespaces-details

Comment (antkmsft) [reply docs/cpp/introduction.md:178]: Thank you! - #6173

**Pull request merged**